deepchem/models/torch_models/flows.py

```diff
@@ -0,0 +1,313 @@
+""" This module contains the implementation of the various flow layers and models"""
+import numpy as np
+import torch
+import torch.nn as nn
+from typing import Optional, Sequence, Tuple, Union
+
+
+class Flow(nn.Module):
+    """
+    Generic class for flow functions
+    """
+
+    def __init__(self):
+        """Initializes the flow function
+        """
+        super().__init__()
+
+    def forward(self, z: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        Forward pass of the flow
+
+        Parameters
+        ----------
+        z: torch.Tensor
+            Input tensor
+
+        Returns
+        -------
+        z_: torch.Tensor
+            Transformed tensor
+        log_det: torch.Tensor
+            Logarithm of the determinant of the Jacobian of the transformation
+        """
+        raise NotImplementedError("Forward pass has not been implemented.")
+
+    def inverse(self, z: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        Inverse pass of the flow
+
+        Parameters
+        ----------
+        z: torch.Tensor
+            Input tensor
+
+        Returns
+        -------
+        z_: torch.Tensor
+            Transformed tensor
+        log_det: torch.Tensor
+            Logarithm of the determinant of the Jacobian of the transformation
+        """
+        raise NotImplementedError("This flow has no algebraic inverse.")
+
+
+# Adding duplicate class here, the layers `Affine` class to be deprecated
```

Comment:
Why are we deprecating that class? We should modify or extend it instead or name this class something different

Reply:
done